src/BoundaryConditions/fill_halo_regions_open.jl

```diff
@@ -1,39 +1,91 @@
-#####
-##### Outer functions for filling halo regions for open boundary conditions.
-#####
-
-# TODO: support true open boundary conditions.
-# For this we need to have separate functions for each of the six boundaries,
-# and need to unroll a loop over the boundary normal direction.
-# The syntax for `getbc` is also different for OpenBoundaryCondition than for others,
-# because the boundary-normal index can vary (and array boundary conditions need to be
-# 3D in general).
-
-@kernel function set_west_or_east_u!(u, i_boundary, bc, grid, args) 
+function fill_bounded_wall_normal_halo_regions!(velocities, clock, fields)
+    grid = velocities.u.grid
+
+    TX, TY, TZ = topology(grid)
+
+    if TX() isa Bounded
+        fill_u_open_halo_regions!(velocities.u, grid, clock, fields)
+    end
+
+    if TY() isa Bounded
+        fill_v_open_halo_regions!(velocities.v, grid, clock, fields)
+    end
+
+    if TZ() isa Bounded
+        fill_w_open_halo_regions!(velocities.w, grid, clock, fields)
+    end
+
+    return nothing
+end
+
+function fill_u_open_halo_regions!(u, grid, args...)
+    arch = architecture(grid)
+
+    west_bc = u.boundary_conditions.west
+    east_bc = u.boundary_conditions.east
+
+    launch!(arch, grid, :yz, _fill_west_and_east_open_halo!, u, west_bc, east_bc, instantiated_location(u), grid, args)
+end
```

Comment:
```suggestion
    launch!(arch, grid, :yz, _fill_west_and_east_open_halo!, u, west_bc, east_bc, instantiated_location(u), grid, args)

    return nothing
end
```

Every function has to end with `return`, that's our convention.

Reply:
you could also write `return launch!` if indeed you are trying to return the output of `launch!`.